runtime-modules/storage/src/data_directory.rs

```diff
@@ -204,14 +212,21 @@ decl_event! {
     pub enum Event<T> where
         <T as Trait>::ContentId,
         StorageObjectOwner = StorageObjectOwner<MemberId<T>, ChannelId, DAOId>,
-        StorageProviderId = StorageProviderId<T>
+        StorageProviderId = StorageProviderId<T>,
+        MultiContent = Vec<ContentParameters<<T as Trait>::ContentId, <T as data_object_type_registry::Trait>::DataObjectTypeId>>,
     {
         /// Emits on adding of the content.
         /// Params:
         /// - Id of the relationship.
-        /// - Id of the member.
+        /// - StorageObjectOwner enum.
         ContentAdded(ContentId, StorageObjectOwner),
 
+        /// Emits on atomic adding of the content.
+        /// Params:
+        /// - StorageObjectOwner enum.
+        /// - MultiContent parameters representation.
+        ContentMultiAdded(StorageObjectOwner, MultiContent),
```

Comment:
For consistency with `ContentAdded` event perhaps `ContentMultiAdded(Vec<ContentId>, StorageObjectOwner)` instead?

Or combine both events into a single `ContentAdded(Vec<ContentParameters>, StorageObjectOwner)` I don't think we need a distinct event to distinguish between multi and single uploads

Reply:
`Or combine both events into a single ContentAdded(Vec<ContentParameters>, StorageObjectOwner) I don't think we need a distinct event to distinguish between multi and single uploads`

I prefer to go with this approach. Done here https://github.com/Joystream/joystream/commit/ae0e77f2a511ab24bd7cc07e689d20b8bfa87d74